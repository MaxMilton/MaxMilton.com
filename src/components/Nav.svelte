<script>
  export let segment;
</script>

<style type="text/postcss">
  @import 'import.css';

  nav {
    padding: 0.3em 1em;
    border-bottom: 1px solid color-mod($dark3 alpha(0.1));
  }

  /* stylelint-disable-next-line selector-max-id */
  #logo {
    color: $text-color;
    font-size: $logo-text-size;
    text-decoration: none;
  }

  .nav-item {
    margin-left: 1em;
    padding: 0.65em 1em;
    text-decoration: none;
    border-bottom: 2px solid transparent;

    &:hover,
    &:focus,
    &:active,
    [aria-current] {
      border-bottom-color: $intent-primary;
    }
  }
</style>

<nav class="dfc">
  <a id="logo" href="/">Max Milton</a>
  <!-- Hide "Home" menu item on small screens to prevent overflow -->
  <a class="dn ns-dib nav-item" aria-current="{segment === undefined ? 'page' : undefined}" href="/">Home</a>
  <a class="nav-item" aria-current="{segment === 'about' ? 'page' : undefined}" href="about">About</a>
  <a rel=prefetch class="nav-item" aria-current="{segment === 'blog' ? 'page' : undefined}" href="blog">Blog</a>
</nav>
