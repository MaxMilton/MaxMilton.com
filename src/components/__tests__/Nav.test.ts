import Nav from '../Nav.svelte';

describe('Nav component', () => {
  it('renders correctly with no props', () => {
    expect.assertions(3);
    function wrapper(): void {
      const target = document.createElement('div');
      new Nav({ target });
      expect(target.querySelector('nav')).not.toBeNull();
      expect(target.innerHTML).toMatchSnapshot();
    }
    expect(wrapper).not.toThrow();
  });
});
