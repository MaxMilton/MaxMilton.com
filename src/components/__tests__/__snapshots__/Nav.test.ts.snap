// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly with no props 1`] = `
"<nav class=\\"dfc\\"><a id=\\"logo\\" href=\\"/\\">Max Milton</a>
  
  <a class=\\"dn ns-dib nav-item\\" aria-current=\\"page\\" href=\\"/\\">Home</a>
  <a class=\\"nav-item\\" href=\\"/about\\">About</a>
  <a rel=\\"prefetch\\" class=\\"nav-item\\" href=\\"/blog\\">Blog</a></nav>"
`;
